coffee_cmd/src/main.rs

```diff
@@ -58,19 +58,34 @@ async fn main() -> Result<(), CoffeeError> {
             coffee_term::show_list(remotes)
         }
         CoffeeCommand::Upgrade { repo, verbose } => {
+            let spinner = if !verbose {
+                Some(term::spinner("Upgrading"))
+            } else {
+                None
+            };
             match coffee.upgrade(&repo, verbose).await {
-                Ok(res) => match res.status {
-                    UpgradeStatus::UpToDate => {
-                        term::info!("Remote repository `{}` is already up to date!", res.repo)
+                Ok(res) => {
+                    if let Some(spinner) = spinner {
+                        spinner.finish();
                     }
```

Comment:
I guess you can do, `spinner.and_then(|splinner| splinner.finish())` not sure that this will work at the first try but we can iterate over it

Reply:
https://github.com/coffee-tools/coffee/pull/217/commits/2099316ecc735c0a871120c668267a4888173766